src/cpp/Agent.cpp

```diff
@@ -36,14 +31,193 @@ static dds::xrce::ClientKey raw_to_clientkey(uint32_t key)
+    topic.representation().object_reference(rep.ref);
+    object_variant.topic(topic);
+}
+
+template<>
+void fill_object_variant<Agent::PUBLISHER_OBJK>(
+        uint16_t participant_id,
+        XmlRep rep,
+        dds::xrce::ObjectVariant& object_variant)
+{
+    dds::xrce::OBJK_PUBLISHER_Representation publisher;
+    publisher.participant_id(XRCEObject::raw_to_objectid(participant_id));
+    publisher.representation().string_representation(rep.xml);
+    object_variant.publisher(publisher);
+}
+
+template<>
+void fill_object_variant<Agent::SUBSCRIBER_OBJK>(
+        uint16_t participant_id,
+        XmlRep rep,
+        dds::xrce::ObjectVariant& object_variant)
+{
+    dds::xrce::OBJK_SUBSCRIBER_Representation subscriber;
+    subscriber.participant_id(XRCEObject::raw_to_objectid(participant_id));
+    subscriber.representation().string_representation(rep.xml);
+    object_variant.subscriber(subscriber);
+}
+
+template<>
+void fill_object_variant<Agent::DATAWRITER_OBJK>(
+        uint16_t publisher_id,
+        XmlRep rep,
+        dds::xrce::ObjectVariant& object_variant)
+{
+    dds::xrce::DATAWRITER_Representation datawriter;
+    datawriter.publisher_id(XRCEObject::raw_to_objectid(publisher_id));
+    datawriter.representation().xml_string_representation(rep.xml);
+    object_variant.data_writer(datawriter);
+}
+
+template<>
+void fill_object_variant<Agent::DATAWRITER_OBJK>(
+        uint16_t publisher_id,
+        RefRep rep,
+        dds::xrce::ObjectVariant& object_variant)
+{
+    dds::xrce::DATAWRITER_Representation datawriter;
+    datawriter.publisher_id(XRCEObject::raw_to_objectid(publisher_id));
+    datawriter.representation().object_reference(rep.ref);
+    object_variant.data_writer(datawriter);
+}
+
+template<>
+void fill_object_variant<Agent::DATAREADER_OBJK>(
+        uint16_t subscriber_id,
+        XmlRep rep,
+        dds::xrce::ObjectVariant& object_variant)
+{
+    dds::xrce::DATAREADER_Representation datareader;
+    datareader.subscriber_id(XRCEObject::raw_to_objectid(subscriber_id));
+    datareader.representation().xml_string_representation(rep.xml);
+    object_variant.data_reader(datareader);
+}
+
+template<>
+void fill_object_variant<Agent::DATAREADER_OBJK>(
+        uint16_t subscriber_id,
+        RefRep rep,
+        dds::xrce::ObjectVariant& object_variant)
+{
+    dds::xrce::DATAREADER_Representation datareader;
+    datareader.subscriber_id(XRCEObject::raw_to_objectid(subscriber_id));
+    datareader.representation().object_reference(rep.ref);
+    object_variant.data_reader(datareader);
+}
+
+template<Agent::ObjectKind object_kind, typename U, typename T>
+bool create_object(
+        uint32_t client_key,
+        uint16_t raw_id,
+        T parent_id,
+        U rep,
+        uint8_t flag,
+        Agent::OpResult& op_result)
+{
+    bool rv = false;
+    Root& root = Root::instance();
+
+    if (std::shared_ptr<ProxyClient> client = root.get_client(raw_to_clientkey(client_key)))
```

Comment:
Use directly `Root::instance()` and avoid one line of code for next readings :D

Reply:
Fixed.